app/src/main/java/app/revanced/integrations/patches/MinimizedPlaybackPatch.java

```diff
@@ -2,12 +2,39 @@
 
 import app.revanced.integrations.shared.PlayerType;
 
+@SuppressWarnings("unused")
 public class MinimizedPlaybackPatch {
 
-    public static boolean isPlaybackNotShort() {
-        return !PlayerType.getCurrent().isNoneHiddenOrSlidingMinimized();
+    /**
+     * Injection point.
+     */
+    public static boolean playbackIsNotShort() {
+        // Steps to verify most edge cases:
+        // 1. Open a regular video
+        // 2. Minimize app (PIP should appear)
+        // 3. Reopen app
+        // 4. Open a Short (without closing the regular video)
+        //    (try opening both Shorts in the video player suggestions AND Shorts from the home feed)
+        // 5. Minimize the app (PIP should not appear)
+        // 6. Reopen app
+        // 7. Close the Short
+        // 8. Resume playing the regular video
+        // 9. Minimize the app (PIP should appear)
+
+        if (!VideoInformation.lastVideoIdIsShort()) {
+            return true; // Definitely is not a Short.
+        }
+
+        // Might be a Short, or might be a prior regular video on screen again after a Short was closed.
+        // This incorrectly prevents PIP if player is in WATCH_WHILE_MINIMIZED after closing a Short,
+        // But there's no way around this unless an additional hook is added to definitively detect
```

Comment:
This may be achievable by checking for a specific component only visible on the screen when Shorts are playing for the future.